CRM/Sqltasks/Action/CreateActivity.php

```diff
@@ -103,13 +111,18 @@ public function createMassActivity() {
       }
     }
     $activity = civicrm_api3('Activity', 'create', $activity_data);
+    $activity_id = (int) $activity['id'];
 
     $excludeSql = '';
     if ($this->_columnExists($contact_table, 'exclude')) {
       $excludeSql = 'AND (exclude IS NULL OR exclude != 1)';
       $this->log('Column "exclude" exists, might skip some rows');
     }
 
+    if ($store_activity_ids) {
+      CRM_Core_DAO::executeQuery("UPDATE `$contact_table` SET `activity_id` = $activity_id");
```

Comment:
We should add `$excludeSql` to this query so rows with `exclude=1` (if the column exists) are not set to that activity ID. Rows with `exclude=1` are handled separately in `createExclusionActivity()` and will get their own activities.

`createExclusionActivity()` needs to be adjusted to add activity IDs for rows with `exclude=1`.